xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/BaseSearchResult.java

```diff
@@ -282,6 +290,23 @@
                     pageLink.setRel(Relations.PAGE);
                     searchResult.getLinks().add(pageLink);
 
+                    Hierarchy hierarchy = new Hierarchy();
+                    for (EntityReference entityReference : doc.getDocumentReference().getReversedReferenceChain()) {
+                        HierarchyItem hierarchyItem = new HierarchyItem();
+                        Document document = xwikiApi.getDocument(entityReference);
+                        if (entityReference instanceof SpaceReference || entityReference instanceof DocumentReference) {
```

Comment:
We usually check the entity type:

    entityReference.getType() == EntityType.SPACE